### ISSUE TITLE
TheoreticalSpectrumGenerator has inefficient parameter access

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#1407]
many times param objects are queried instead of a member variable (that are updated via updateMemberVariables_)
e.g. it should not use param_.getValue("add_first_prefix_ion")

**Comment by liangoaix** [comment]
Should we define a new member variable only here? Because I found this schema was used in many cases.

**Comment by timosachsenberg** [comment]
best would be to fix this everywhere where param is used.
The problem is that there are some cases that probably should never have used param to set and retrieve parameters.
In this case I think it is sufficient to do this for TheoreticalSpectrumGenerator

**Comment by cbielow** [comment]
Agreeing with Timo.

Also, in 
```
void TheoreticalSpectrumGenerator::addAbundantImmoniumIons(RichPeakSpectrum & spec) const
```
There are hardcoded ion masses with probably insufficient precision -- at least for high-res applications.
While you're at it, could you please replace these with static EmpiricalFormulas and get the masses from there?! Thx :)

**Comment by timosachsenberg** [comment]
Immonium ions    FragmentComposition (+H+)  Mass (+H+)
iY C8H10NO 136.0762
iW C10H11N2 159.0922
iF C8H10N 120.0813
iH C5H8N3 110.0718
iC C2H6NS 76.0221
iP C4H8N 70.0656

**Comment by liangoaix** [comment]
Sure, I will fix it.

**Comment by timosachsenberg** [comment]
I have the information not at hand for iL/iI

**Comment by liangoaix** [comment]
http://www.abrf.org/researchgroups/massspectrometry/eposters/ms97quiz/residuemasses.html

**Comment by timosachsenberg** [comment]
These are more accurate but still miss the composition

**Comment by liangoaix** [comment]
Is it always losing -COOH? then 
iL C5H12N 86.09698

**Comment by timosachsenberg** [comment]
I think COOH2 is lost and one proton gained so pretty close to COOH:
https://www.researchgate.net/profile/Talat_Yalcin/publication/227874634_Pathways_to_Immonium_Ions_in_the_Fragmentation_of_Protonated_Peptides/links/0912f50c9b6e64fb37000000.pdf

**Comment by liangoaix** [comment]
Now it's clear, thanks.

**Comment by hendrikweisser** [comment]
Parameter access was fixed in #1751.
